### PR TITLE
[DNM BECAUSE IT CAUSES A CRASH] nautilus: libcephfs: ignore restoring the open files limit 

### PR DESCRIPTION
**Pull request by smithfarm**
backport tracker: https://tracker.ceph.com/issues/47020

---

backport of https://github.com/ceph/ceph/pull/36515
parent tracker: https://tracker.ceph.com/issues/43039

this backport was staged using ceph-backport.sh version 15.1.1.389
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by smithfarm** [comment]
```
/home/jenkins-build/build/workspace/ceph-pull-requests/qa/standalone/ceph-helpers.sh:462: run_mon:  ceph-mon --id a --osd-failsafe-full-ratio=.99 --mon-osd-full-ratio=.99 --mon-data-avail-crit=1 --mon-data-avail-warn=5 --paxos-propose-interval=0.1 --osd-crush-chooseleaf-type=0 --debug-mon 20 --debug-ms 20 --debug-paxos 20 --chdir= --mon-data=td/mgr-dashboard-smoke/a '--log-file=td/mgr-dashboard-smoke/$name.log' '--admin-socket=/tmp/ceph-asok.14036/$cluster-$name.asok' --mon-cluster-log-file=td/mgr-dashboard-smoke/log --run-dir=td/mgr-dashboard-smoke '--pid-file=td/mgr-dashboard-smoke/$name.pid' --mon-allow-pool-delete --mon-osd-backfillfull-ratio .99
/home/jenkins-build/build/workspace/ceph-pull-requests/src/msg/async/Event.cc: In function 'int EventCenter::init(int, unsigned int, const string&)' thread 7f04c01933c0 time 2020-09-29 17:49:29.625229
/home/jenkins-build/build/workspace/ceph-pull-requests/src/msg/async/Event.cc: 106: FAILED ceph_assert(nevent == 0)
 ceph version Development (no_version) nautilus (stable)
 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x1aa) [0x7f04b6d0df40]
 2: (()+0x13df1c2) [0x7f04b6d0e1c2]
 3: (EventCenter::init(int, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x4f) [0x7f04b70d5e87]
 4: (NetworkStack::NetworkStack(CephContext*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x27e) [0x7f04b70e70cc]
 5: (PosixNetworkStack::PosixNetworkStack(CephContext*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x2b) [0x7f04b70e2ec1]
 6: (void __gnu_cxx::new_allocator<PosixNetworkStack>::construct<PosixNetworkStack, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(PosixNetworkStack*, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x5f) [0x7f04b70ea6a7]
 7: (void std::allocator_traits<std::allocator<PosixNetworkStack> >::construct<PosixNetworkStack, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(std::allocator<PosixNetworkStack>&, PosixNetworkStack*, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x4a) [0x7f04b70ea3c8]
 8: (std::_Sp_counted_ptr_inplace<PosixNetworkStack, std::allocator<PosixNetworkStack>, (__gnu_cxx::_Lock_policy)2>::_Sp_counted_ptr_inplace<CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(std::allocator<PosixNetworkStack>, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0xbd) [0x7f04b70e9f75]
 9: (std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count<PosixNetworkStack, std::allocator<PosixNetworkStack>, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(std::_Sp_make_shared_tag, PosixNetworkStack*, std::allocator<PosixNetworkStack> const&, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0xe0) [0x7f04b70e9a72]
 10: (std::__shared_ptr<PosixNetworkStack, (__gnu_cxx::_Lock_policy)2>::__shared_ptr<std::allocator<PosixNetworkStack>, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(std::_Sp_make_shared_tag, std::allocator<PosixNetworkStack> const&, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x6e) [0x7f04b70e9774]
 11: (std::shared_ptr<PosixNetworkStack>::shared_ptr<std::allocator<PosixNetworkStack>, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(std::_Sp_make_shared_tag, std::allocator<PosixNetworkStack> const&, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x57) [0x7f04b70e9399]
 12: (std::shared_ptr<PosixNetworkStack> std::allocate_shared<PosixNetworkStack, std::allocator<PosixNetworkStack>, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(std::allocator<PosixNetworkStack> const&, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x52) [0x7f04b70e8e75]
 13: (std::shared_ptr<PosixNetworkStack> std::make_shared<PosixNetworkStack, CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>(CephContext*&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x66) [0x7f04b70e8b87]
 14: (NetworkStack::create(CephContext*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x67) [0x7f04b70e6937]
 15: (StackSingleton::ready(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)+0x50) [0x7f04b7070d3e]
 16: (AsyncMessenger::AsyncMessenger(CephContext*, entity_name_t, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned long)+0x3ed) [0x7f04b7067ddb]
 17: (Messenger::create(CephContext*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, entity_name_t, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned long, unsigned long)+0x1ed) [0x7f04b701544f]
 18: (main()+0x5ce9) [0x55c5542bde51]
 19: (__libc_start_main()+0xe7) [0x7f04b3ae7b97]
 20: (_start()+0x2a) [0x55c5542b474a]
2020-09-29 17:49:29.652 7f04c01933c0 -1 /home/jenkins-build/build/workspace/ceph-pull-requests/src/msg/async/Event.cc: In function 'int EventCenter::init(int, unsigned int, const string&)' thread 7f04c01933c0 time 2020-09-29 17:49:29.625229
/home/jenkins-build/build/workspace/ceph-pull-requests/src/msg/async/Event.cc: 106: FAILED ceph_assert(nevent == 0)
```

**Comment by smithfarm** [comment]
@lxbsz This change is causing us to hit an assert (see previous comment) in `nautilus`. Any idea how to fix it?

**Comment by lxbsz** [comment]
> @lxbsz This change is causing us to hit an assert (see previous comment) in `nautilus`. Any idea how to fix it?

Will check it, but I have no idea what has happened with this change.

**Comment by lxbsz** [comment]
Add one log in the init(), not sure why it called twice.

```

    -5> 2020-09-30 05:46:39.831 7f1e8eb1c3c0 -1 Event(0x55a0d13bb700 nevent=0 time_id=1).init nevent: 0
    -4> 2020-09-30 05:46:39.831 7f1e8eb1c3c0 -1 Event(0x55a0d13bb700 nevent=5000 time_id=1).init nevent: 5000
    -3> 2020-09-30 05:46:39.863 7f1e8eb1c3c0 -1 /home/jenkins-build/build/workspace/ceph-pull-requests/src/msg/async/Event.cc: In function 'int EventCenter::init(int, unsigned int, const string&)' thread 7f1e8eb1c3c0 time 2020-09-30 05:46:39.836699
/home/jenkins-build/build/workspace/ceph-pull-requests/src/msg/async/Event.cc: 107: FAILED ceph_assert(nevent == 0)

```

**Comment by lxbsz** [comment]
@smithfarm For this you can also just remove "msg: throw a system error when center.init fails " commit, it just one debug commit and doesn't make much sense to fixing this issue.

**Comment by batrick** [comment]
Will have @lxbsz pick this up.

File: src/msg/async/Stack.cc [hunk 1]
```diff
@@ -118,6 +118,9 @@ NetworkStack::NetworkStack(CephContext *c, const string &t): type(t), started(fa
   for (unsigned i = 0; i < num_workers; ++i) {
     Worker *w = create_worker(cct, type, i);
     w->center.init(InitEventNumber, i, type);
```
Comment (lxbsz) [inline src/msg/async/Stack.cc:120]: Okay, I think you just forgot to remove this line.

File: src/msg/async/Stack.cc [hunk 2]
```diff
@@ -118,6 +118,9 @@ NetworkStack::NetworkStack(CephContext *c, const string &t): type(t), started(fa
...
+    int ret = w->center.init(InitEventNumber, i, type);
```
Comment (lxbsz) [inline src/msg/async/Stack.cc:121]: And doing the `center.init()` twice here.

**Pull request closed without merging**